### ISSUE TITLE
how to switch coordinate system for labels in nus-mono dataset from cam to lidar

### ISSUE THREAD

**Issue by BoomSky0416** [open-mmlab/mmdetection3d#1582]
I tried to get the bbox_3d in lidar coordinate system based on labels in nus-mono dataset(nuscenes_infos_train_mono3d.coco.json). I tried to transform the bbox_3d in cam based on corresponding cam2lidar info stored in the [nuscenes_infos_train.pkl]. And I found the first info in [nuscenes_infos_train.pkl] corresponds to the first 6 infos in [nuscenes_infos_train_mono3d.coco.json]. 
Here is code I use to transform the bbox_3d in cam. Only the first and the third transformed cam bbox can be found in gt_boxes.

```
import mmcv
from mmdet3d.core.bbox import CameraInstance3DBoxes, get_box_type, LiDARInstance3DBoxes
import numpy as np
import torch

mono_data = mmcv.load('/workspace/dataset/public/algorithm-datasets/nuscenes/nuscenes_infos_train_mono3d.coco.json')
multi_data = mmcv.load('/workspace/dataset/public/algorithm-datasets/nuscenes/nuscenes_infos_train.pkl')
camera_name_list = ['CAM_FRONT', 'CAM_FRONT_RIGHT', 'CAM_FRONT_LEFT', 'CAM_BACK', 'CAM_BACK_LEFT', 'CAM_BACK_RIGHT']
print('total:', len(mono_data['images']))

arr_list = []
multi_info = multi_data['infos'][0]
gt_boxes = torch.tensor(multi_info['gt_boxes'])

for i in range(6):
    img_info = mono_data['images'][i]
    img_anno = mono_data['annotations'][i]
    single_image_box = np.array(img_anno['bbox_cam3d']).reshape(1,-1)

    
    cam2lidar_rt = np.eye(4)
    cam2lidar_rt[:3, :3] = multi_info['cams'][camera_name_list[i%6]]['sensor2lidar_rotation'].T
    cam2lidar_rt[3, :3] = multi_info['cams'][camera_name_list[i%6]]['sensor2lidar_translation']

    arr = torch.tensor(single_image_box)
    extended_xyz = torch.cat(
        [arr[..., :3], arr.new_ones(arr.size(0), 1)], dim=-1)

    rt_mat = arr.new_tensor(cam2lidar_rt)
    rot_sin = rt_mat[2, 0]
    rot_cos = rt_mat[0, 0]
    angle = np.arctan2(rot_sin, rot_cos)
    xyz = extended_xyz @ rt_mat
    x_size, y_size, z_size = arr[..., 3:4], arr[..., 4:5], arr[..., 5:6]
    xyz_size = torch.cat([z_size, x_size, y_size], dim=-1)
    yaw = arr[..., 6:7]
    yaw += angle
    yaw -= np.pi/2
    arr = torch.cat([xyz[..., :3], xyz_size, yaw], dim=-1)
    arr_list.append(arr)
print('transformed cam bbox:',torch.cat(arr_list,0))
print('lidar gt:',gt_boxes)
```

Here is the printed result.

```
total: 168780
tensor([[-1.0228e+01,  1.9461e+01,  3.7436e-02,  2.3120e+00,  7.5160e+00,
          3.0930e+00, -9.6429e-01],
        [ 2.1802e+01,  2.6459e+00, -5.6169e-01,  2.3120e+00,  7.5160e+00,
          3.0930e+00,  9.7850e-01],
        [-4.7528e+01,  3.5137e+01,  6.9496e-01,  2.8770e+00,  6.3720e+00,
          2.9780e+00, -4.1602e+00],
        [-9.6379e+00, -3.0895e+01, -8.0711e-01,  2.1560e+00,  6.2270e+00,
          2.6010e+00, -3.2456e+00],
        [-1.1952e+01, -4.2121e+00, -1.4657e+00,  1.8710e+00,  4.4780e+00,
          1.4560e+00, -1.4234e+00],
        [ 1.3400e+01, -9.0306e+00, -1.3297e+00,  3.0000e-01,  2.9100e-01,
          7.3400e-01, -1.0839e+00]], dtype=torch.float64)
tensor([[-1.6184e+01, -1.1740e+00, -1.2405e+00,  3.0000e-01,  2.9100e-01,
          7.3400e-01,  1.3645e+00],
        [-1.5449e+01, -4.2877e+00, -1.3014e+00,  3.1500e-01,  3.3800e-01,
          7.1200e-01,  1.2599e+00],
        [-1.0228e+01,  1.9461e+01,  3.7436e-02,  2.3120e+00,  7.5160e+00,
          3.0930e+00, -9.6416e-01],
        [ 9.2144e+00, -5.5796e+00, -1.0786e+00,  1.6380e+00,  4.2500e+00,
          1.4400e+00, -1.9293e+00],
        [-1.5727e+01, -8.1609e-01, -6.9794e-01,  7.3900e-01,  5.6300e-01,
          1.7110e+00,  1.3645e+00],
        [ 3.8465e-01, -1.3228e+01, -1.2146e+00,  1.8710e+00,  4.4780e+00,
          1.4560e+00, -2.6640e+00],
        [-4.7528e+01,  3.5137e+01,  6.9496e-01,  2.8770e+00,  6.3720e+00,
          2.9780e+00, -4.1599e+00],
        [-1.6106e+01, -7.1648e-02, -6.8628e-01,  6.6500e-01,  5.4400e-01,
          1.7390e+00,  1.3645e+00],
        [-1.5941e+01, -2.4479e+00, -1.2858e+00,  3.3800e-01,  3.0900e-01,
          7.1200e-01,  1.3580e+00],
        [-1.9383e+01,  2.5539e+01,  3.1919e-02,  2.1560e+00,  6.2270e+00,
          2.6010e+00, -1.0600e+00]], dtype=torch.float64)
```
Anyone helps? What's wrong in the coordinate transformation process? Thanks!

**Comment by BoomSky0416** [comment]
Problem solved. Close here.

**Comment by djiajunustc** [comment]
@BoomSky0416 Could you please tell me how to switch it?
Maybe by using BoxMode.convert and extrinsic.T ?
